### PR TITLE
handle unlocked bootloader and enabled oem unlocking

### LINKED ISSUE #2: Handle unlocked bootloader status
- if the bootloader is unlocked, prompt the user to lock it before starting setup. It's not possible to lock the bootloader from the OS, device has to be rebooted to bootloader interface for that. Prompt should strongly encourage the user to lock the bootloader, skipping the prompt should require double confirmation from the user via a second prompt.
- if OEM unlocking is enabled, disable it automatically before starting setup. This doesn't require rebooting, calling `OemLockManager#setOemUnlockAllowedByUser(false)` should be enough

UX

Simply show a dialog when "Start" button is tapped (if bootloader is unlocked at that point).
also display relevant info in the welcome screen in orange/red color (additionally). so that the dialog doesn't come up as a surprise.

### PR DESCRIPTION
**Pull request by jiteshsingh**
shows prompts to the user for locking bootloader and disabling
oem unlocking

fixes: https://github.com/GrapheneOS/platform_packages_apps_SetupWizard2/issues/2

### CODE REVIEW

File: res/values/strings.xml [hunk 1]
```diff
@@ -36,4 +36,16 @@
     <string name="swipe_gestures_title">Swipe to navigate your phone</string>
     <string name="swipe_gestures_desc">Learn gestures to go home, go back and switch between apps</string>
     <string name="try_it">Try it</string>
+    <string name="oem_unlocked_warning">The boot loader is unlocked and software integrity can not be guaranteed. Any data stored on this device may be available to attackers. Do not store any sensitive data on this device.</string>
```
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
    <string name="oem_unlocked_warning">The boot loader is unlocked and software integrity can not be guaranteed. Any data stored on this device may be available to attackers. Do not store any sensitive data on this device.</string>
```

Aside from wording improvements in these strings which we've been discussing, I am not certain that it makes sense to label strings having to do with bootloader lock/unlock status as "oem_unlocked" as OEM unlocking is separate from bootloader locking/unlocking, which can be confusing.

Comment (jiteshsingh) [reply res/values/strings.xml]: the underlying api uses this terminology: https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/service/oemlock/OemLockManager.java#155

the api for determining bootloader unlocked status is called `OemLockManager.isDeviceOemUnlocked()`.

so it can be looked at this way:
- oem unlocked
- oem unlocking enabled

the above 2 are different things (as per upstream terminology)

Comment (matchboxbananasynergy) [reply res/values/strings.xml]: I see, in that case it likely makes sense to leave it as is. Thanks for the explanation!

File: AndroidManifest.xml [hunk 2]
```diff
@@ -5,7 +5,9 @@
     tools:ignore="ProtectedPermissions">
 
     <uses-permission android:name="android.permission.CHANGE_CONFIGURATION" />
+    <uses-permission android:name="android.permission.MANAGE_USER_OEM_UNLOCK_STATE" />
```
Comment (quh4gko8) [inline AndroidManifest.xml:8]: Any privileged permissions added here requires to be added at privapp whitelist/allowlist file

**Review by quh4gko8 — CHANGES_REQUESTED** [review]

File: AndroidManifest.xml (hunk 2 above)
Comment (jiteshsingh) [reply AndroidManifest.xml:8]: added

File: java/app/grapheneos/setupwizard/view/activity/WelcomeActivity.kt [hunk 3]
```diff
@@ -1,20 +1,18 @@
...
 import android.widget.TextView
 import androidx.annotation.MainThread
 
-import com.google.android.setupcompat.template.FooterButtonStyleUtils
 import com.google.android.setupcompat.util.WizardManagerHelper
-import com.google.android.setupdesign.GlifLayout
 
 import app.grapheneos.setupwizard.R
 import app.grapheneos.setupwizard.action.FinishActions
-import app.grapheneos.setupwizard.action.SetupWizard
```
Comment (quh4gko8) [inline java/app/grapheneos/setupwizard/view/activity/WelcomeActivity.kt]: This is needed in #20, do not remove this import. Import cleanups are usually done in separate commits

**Review by quh4gko8 — CHANGES_REQUESTED** [review]

File: java/app/grapheneos/setupwizard/view/activity/WelcomeActivity.kt (hunk 3 above)
Comment (quh4gko8) [reply java/app/grapheneos/setupwizard/view/activity/WelcomeActivity.kt]: Particularly, the import for `SetupWizard` object

File: res/values/strings.xml [hunk 4]
```diff
@@ -36,4 +36,16 @@
...
+    <string name="visit_this_link_on_another_device">Visit this link on another device:</string>
+    <string name="oem_lock_url" translatable="false">g.co/ABH</string>
+    <string name="oem_unlocked_device_setup_confirmation">Boot loader is unlocked which renders the device insecure as displayed in the warning. Do you still want to continue setting up?</string>
```
Comment (flawedworld) [inline res/values/strings.xml]: This text should refer the user back to the installation steps on grapheneos.org

**Comment by flawedworld** [comment]
My _personal_ preference would be that we do not allow the user to progress at all without doing something like 5 taps on a place on the screen etc, we should reach consensus on this IMHO. People have a nasty habit of flicking through screens and ignoring the words on them.

**Comment by matchboxbananasynergy** [comment]
> My _personal_ preference would be that we do not allow the user to progress at all without doing something like 5 taps on a place on the screen etc, we should reach consensus on this IMHO. People have a nasty habit of flicking through screens and ignoring the words on them.

This was considered, but the hope is that seeing the warning on the first screen, and then seeing it again when you press next is enough. Providing live support in the rooms will be a good "test" to gauge cases where those things are still not enough. If we realize that many people **unwillingly** still walk around with an unlocked bootloader due to lack of reading, then another approach should be taken to ensure people aren't shooting themselves in the foot.

Another approach that could be taken in addition to these steps in the setup wizard is to surface a notification about the bootloader being unlocked. When clicked, it could take you to a page explaining the situation, why it matters, and that locking will require wiping everything, in order to prepare them.

File: res/values/strings.xml (hunk 4 above)
Comment (jiteshsingh) [reply res/values/strings.xml]: yes this is pending actually, particularly this is why I've still kept the PR as draft.
there were different opinions regarding this on whether to just refer to existing installation steps or have a dedicated page and shorter url.

File: java/app/grapheneos/setupwizard/view/activity/WelcomeActivity.kt (hunk 3 above)
Comment (jiteshsingh) [reply java/app/grapheneos/setupwizard/view/activity/WelcomeActivity.kt]: done

File: res/values/strings.xml (hunk 1 above)
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
    <string name="oem_unlocked_warning">The bootloader is unlocked and software integrity cannot be guaranteed. Any data stored on this device may be available to attackers. Do not store any sensitive data on this device.</string>
```

File: res/values/strings.xml [hunk 5]
```diff
@@ -36,4 +36,16 @@
     <string name="swipe_gestures_title">Swipe to navigate your phone</string>
     <string name="swipe_gestures_desc">Learn gestures to go home, go back and switch between apps</string>
     <string name="try_it">Try it</string>
+    <string name="oem_unlocked_warning">The boot loader is unlocked and software integrity can not be guaranteed. Any data stored on this device may be available to attackers. Do not store any sensitive data on this device.</string>
+    <string name="visit_this_link_on_another_device">Visit this link on another device:</string>
```
Comment (matchboxbananasynergy) [inline res/values/strings.xml:42]:
```suggestion
    <string name="visit_this_link_on_another_device">To correct this, visit this link on another device and tap "Next" on this screen:</string>
```

Comment (matchboxbananasynergy) [reply res/values/strings.xml:42]: What do we think about the wording here? Trying to make the flow as easy to understand as possible.

File: res/values/strings.xml (hunk 4 above)
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
    <string name="oem_unlocked_device_setup_confirmation">The bootloader is unlocked which renders the device insecure. It is highly recommended to correct this. Locking the bootloader will wipe all user data. Do you still want to continue setting up without locking?</string>
```

Comment (matchboxbananasynergy) [reply res/values/strings.xml]: I am unsure of the wording here. My intention is to make it so it is understood that locking the bootloader isn't a trivial thing you can do at a later time, as locking wipes everything, so they should *really* do it now rather than proceeding.

File: res/values/strings.xml [hunk 6]
```diff
@@ -36,4 +36,16 @@
...
+    <string name="oem_lock_url" translatable="false">g.co/ABH</string>
+    <string name="oem_unlocked_device_setup_confirmation">Boot loader is unlocked which renders the device insecure as displayed in the warning. Do you still want to continue setting up?</string>
+    <string name="confirmation">Confirmation</string>
+    <string name="yes_continue">Yes, Continue</string>
```
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
    <string name="yes_continue">Yes, continue</string>
```

File: res/values/strings.xml [hunk 7]
```diff
@@ -36,4 +36,16 @@
...
+    <string name="no_reboot_to_bootloader">No, Reboot to bootloader</string>
```
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
    <string name="no_reboot_to_bootloader">No, reboot to bootloader</string>
```

File: res/values/strings.xml [hunk 8]
```diff
@@ -36,4 +36,16 @@
...
+    <string name="yes_disable_now">Yes, Disable now</string>
```
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
    <string name="yes_disable_now">Yes, disable now</string>
```

File: res/values/strings.xml [hunk 9]
```diff
@@ -36,4 +36,16 @@
...
+    <string name="no_disable_later">No, I will disable later</string>
```
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
    <string name="no_disable_later">No, I will disable it later</string>
```

File: res/values/strings.xml [hunk 10]
```diff
@@ -36,4 +36,16 @@
...
+    <string name="oem_unlock_disabling_suggestion">OEM unlocking is enabled, we recommend disabling.
```
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
    <string name="oem_unlock_disabling_suggestion">OEM unlocking is enabled, it is recommended to disable this.
            Would you like to disable it now?
        You can also disable it later from developer options in the Settings app.</string>
```

File: res/values/strings.xml [hunk 11]
```diff
@@ -36,4 +36,16 @@
     <string name="swipe_gestures_title">Swipe to navigate your phone</string>
     <string name="swipe_gestures_desc">Learn gestures to go home, go back and switch between apps</string>
     <string name="try_it">Try it</string>
+    <string name="oem_unlocked_warning">The boot loader is unlocked and software integrity can not be guaranteed. Any data stored on this device may be available to attackers. Do not store any sensitive data on this device.</string>
+    <string name="visit_this_link_on_another_device">Visit this link on another device:</string>
+    <string name="oem_lock_url" translatable="false">g.co/ABH</string>
```
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
    <string name="oem_lock_url" translatable="false">grapheneos.org/UBL</string>
```

**Comment by matchboxbananasynergy** [comment]
@jiteshsingh Let's also make sure that both of these only apply to setup in Owner, not secondary or guest profiles.

File: res/values/strings.xml (hunk 1 above)
Comment (jiteshsingh) [reply res/values/strings.xml]: explained here: https://github.com/GrapheneOS/platform_packages_apps_SetupWizard2/pull/19#discussion_r1501694465
it's not confusing if we adopt the terminolgy upstream uses (also necessary for code consistency).
although for user facing text, we can do whatever feels good for the UX.

File: res/values/strings.xml (hunk 4 above)
Comment (jiteshsingh) [reply res/values/strings.xml]: an alternative I suggest:

_The bootloader is unlocked which renders the device insecure as displayed in the warning. Do you still want to continue setting up?
Please note that locking bootloader later will wipe all your data, so we recommend rebooting to bootloader interface and locking it now._

File: res/values/strings.xml (hunk 9 above)
Comment (jiteshsingh) [reply res/values/strings.xml]: The label is becoming long for a button. although I'll apply the suggestion regardless.

File: res/values/strings.xml (hunk 5 above)
Comment (jiteshsingh) [reply res/values/strings.xml:42]: not sure, because it'll read like this: 

_.... and tap "Next" on this screen: grapheneos.org/UBL_

actually the second confirmation on next button is just another blocker, meant to further warn the user (although it hosts the handy button for booting into fastboot).
if we want, we can additionally include the button ~just below the warning~ in main screen.

File: res/values/strings.xml (hunk 10 above)
Comment (jiteshsingh) [reply res/values/strings.xml]: I think it's fine and unambiguous. expanding text too much for a dialog is not a good idea imo.

**Comment by jiteshsingh** [comment]
> @jiteshsingh Let's also make sure that both of these only apply to setup in Owner, not secondary or guest profiles.

done

File: res/values/strings.xml (hunk 4 above)
Comment (jiteshsingh) [reply res/values/strings.xml]: resolved now

File: res/values/strings.xml (hunk 5 above)
Comment (jiteshsingh) [reply res/values/strings.xml:40]: here's the UX I suggest for this intended action.

Android-Screenshot-from-2024-46-27-23-46-29-693873170

File: java/app/grapheneos/setupwizard/action/SetupWizard.kt [hunk 12]
```diff
@@ -95,7 +95,11 @@ object SetupWizard {
         return PartnerConfigHelper.getSuwDefaultThemeString(appContext)
     }
 
+    // primary user here means the first user of the device, the user that is implicitly created
+    // on first boot.
+    // any other subsequent users are secondary users
     val isPrimaryUser: Boolean by lazy {
         appContext.getSystemService(UserManager::class.java).isSystemUser
     }
+    val isSecondaryUser = !isPrimaryUser
```
Comment (muhomorr) [inline java/app/grapheneos/setupwizard/action/SetupWizard.kt]: This declaration makes `lazy` above useless, since `isPrimaryUser` is immediately evaluated.

`UserManager.isSystemUser()` call is cheap anyway, there's no need to load it lazily.

File: java/app/grapheneos/setupwizard/data/WelcomeData.kt [hunk 13]
```diff
@@ -7,6 +7,8 @@ import java.util.Locale
 
 object WelcomeData : ViewModel() {
     val selectedLanguage = MutableLiveData<Locale>()
+    val oemUnlocked = MutableLiveData<Boolean>() // bootloader unlocked
```
Comment (muhomorr) [inline java/app/grapheneos/setupwizard/data/WelcomeData.kt:10]: `oemUnlocked` is not the same as bootloader being unlocked

**Review by muhomorr — CHANGES_REQUESTED** [review]

File: java/app/grapheneos/setupwizard/action/SetupWizard.kt (hunk 12 above)
Comment (jiteshsingh) [reply java/app/grapheneos/setupwizard/action/SetupWizard.kt]: made `isSecondayUser` lazy as well, thanks for pointing out!
yes the call is cheap, and another reason for declaring lazy is to prevent mishappenings due to an out-of-order class loading, although that's very rare, but just in case.

File: java/app/grapheneos/setupwizard/data/WelcomeData.kt (hunk 13 above)
Comment (jiteshsingh) [reply java/app/grapheneos/setupwizard/data/WelcomeData.kt:10]: the api doc seems to say so, although a bit ambiguous; so I've been waiting for test results on real device (don't have a test device at hand).
are you sure this is not the same API we want?

Comment (muhomorr) [reply java/app/grapheneos/setupwizard/data/WelcomeData.kt:10]: I've made a mistake here, oemUnlocked indeed indicates whether the bootloader is unlocked. Got confused by OemLockManager method names.

File: java/app/grapheneos/setupwizard/action/WelcomeActions.kt [hunk 14]
```diff
@@ -89,4 +95,82 @@ object WelcomeActions {
...
+    private fun disableOemUnlockByUser() {
+        getOemLockManager()?.isOemUnlockAllowedByUser = false
+        refreshOemUnlockStatus()
+    }
+
+    private fun getOemLockManager(): OemLockManager? {
+        return appContext.getSystemService(OemLockManager::class.java)
+    }
+
+    private fun rebootBootloader() {
```
Comment (muhomorr) [inline java/app/grapheneos/setupwizard/action/WelcomeActions.kt]: rebootToBootloader would be a better name. Bootloader reboot is a separate thing (when device is in bootloader interface, bootloader itself can be rebooted)

File: java/app/grapheneos/setupwizard/action/WelcomeActions.kt [hunk 15]
```diff
@@ -89,4 +95,82 @@ object WelcomeActions {
...
+    }
+
+    fun next(activity: Activity) {
+        if (Build.isDebuggable()) {
+            // we allow free pass for development features on debug builds of the OS
+            SetupWizard.next(activity)
+            return
+        }
+        if (SetupWizard.isSecondaryUser) {
+            // secondary users should not be bothered for this as they're not admins
```
Comment (muhomorr) [inline java/app/grapheneos/setupwizard/action/WelcomeActions.kt]: Secondary users can be admins too, depending on OS configuration. In any case, I agree that there's no need to show these warnings for them.

File: res/values/colors.xml [hunk 16]
```diff
@@ -1,3 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
+    <color name="oem_unlocked_warning_color">@android:color/holo_orange_dark</color>
```
Comment (muhomorr) [inline res/values/colors.xml]: Holo styling has been long deprecated and will likely be removed at some point.

File: res/values/strings.xml [hunk 17]
```diff
@@ -38,4 +38,16 @@
     <string name="swipe_gestures_title">Swipe to navigate your phone</string>
     <string name="swipe_gestures_desc">Learn gestures to go home, go back and switch between apps</string>
     <string name="try_it">Try it</string>
+    <string name="oem_unlocked_warning">The boot loader is unlocked and software integrity can not be guaranteed. Any data stored on this device may be available to attackers. Do not store any sensitive data on this device.</string>
```
Comment (muhomorr) [inline res/values/strings.xml]: "boot loader" -> "bootloader"

File: java/app/grapheneos/setupwizard/action/WelcomeActions.kt (hunk 14 above)
Comment (jiteshsingh) [reply java/app/grapheneos/setupwizard/action/WelcomeActions.kt]: sounds good

File: res/values/strings.xml (hunk 4 above)
Comment (jiteshsingh) [reply res/values/strings.xml]: handled by the new fullscreen warning.

File: res/values/strings.xml (hunk 5 above)
Comment (jiteshsingh) [reply res/values/strings.xml:42]: handled by the new fullscreen warning.

File: res/values/colors.xml (hunk 16 above)
Comment (jiteshsingh) [reply res/values/colors.xml]: removed reference

File: res/values/strings.xml (hunk 17 above)
Comment (jiteshsingh) [reply res/values/strings.xml]: done

File: java/app/grapheneos/setupwizard/action/WelcomeActions.kt (hunk 15 above)
Comment (jiteshsingh) [reply java/app/grapheneos/setupwizard/action/WelcomeActions.kt]: updated comment

File: res/values/strings.xml [hunk 18]
```diff
@@ -38,4 +38,29 @@
...
+    <string name="oem_unlocked_warning">The bootloader is unlocked and software integrity can not be guaranteed. Any data stored on this device may be available to attackers. Do not store any sensitive data on this device.</string>
+    <string name="visit_this_link_on_another_device">Visit this link on another device:</string>
+    <string name="oem_lock_url" translatable="false">grapheneos.org/UBL</string>
+    <string name="oem_unlocked_device_setup_confirmation">Bootloader is unlocked which renders the device insecure as displayed in the warning. Do you still want to continue setting up?</string>
+    <string name="confirmation">Confirmation</string>
+    <string name="yes_continue">Yes, continue</string>
+    <string name="no_reboot_to_bootloader">No, reboot to bootloader</string>
+    <string name="yes_disable_now">Yes, disable now</string>
+    <string name="no_disable_later">No, I will disable it later</string>
+    <string name="oem_unlock_disabling_suggestion">OEM unlocking is enabled, we recommend disabling.
```
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
    <string name="oem_unlock_disabling_suggestion">OEM unlocking is enabled, we recommend disabling it.
```

**Comment by matchboxbananasynergy** [comment]
When this is merged, this will also need to be merged: https://github.com/GrapheneOS/grapheneos.org/pull/887

Furthermore, a PR redirecting `grapheneos.org/uBL` to the above FAQ section will be needed.

File: res/values/strings.xml (hunk 18 above)
Comment (jiteshsingh) [reply res/values/strings.xml]: applied

File: res/values/strings.xml [hunk 19]
```diff
@@ -38,4 +38,29 @@
     <string name="swipe_gestures_title">Swipe to navigate your phone</string>
     <string name="swipe_gestures_desc">Learn gestures to go home, go back and switch between apps</string>
     <string name="try_it">Try it</string>
+    <string name="oem_unlocked_warning">The bootloader is unlocked and software integrity can not be guaranteed. Any data stored on this device may be available to attackers. Do not store any sensitive data on this device.</string>
+    <string name="visit_this_link_on_another_device">Visit this link on another device:</string>
+    <string name="oem_lock_url" translatable="false">grapheneos.org/UBL</string>
+    <string name="oem_unlocked_device_setup_confirmation">Bootloader is unlocked which renders the device insecure as displayed in the warning. Do you still want to continue setting up?</string>
+    <string name="confirmation">Confirmation</string>
+    <string name="yes_continue">Yes, continue</string>
+    <string name="no_reboot_to_bootloader">No, reboot to bootloader</string>
```
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
    <string name="no_reboot_to_bootloader">Reboot to bootloader</string>
```

File: res/values/strings.xml [hunk 20]
```diff
@@ -38,4 +38,29 @@
     <string name="swipe_gestures_title">Swipe to navigate your phone</string>
     <string name="swipe_gestures_desc">Learn gestures to go home, go back and switch between apps</string>
     <string name="try_it">Try it</string>
+    <string name="oem_unlocked_warning">The bootloader is unlocked and software integrity can not be guaranteed. Any data stored on this device may be available to attackers. Do not store any sensitive data on this device.</string>
+    <string name="visit_this_link_on_another_device">Visit this link on another device:</string>
+    <string name="oem_lock_url" translatable="false">grapheneos.org/UBL</string>
+    <string name="oem_unlocked_device_setup_confirmation">Bootloader is unlocked which renders the device insecure as displayed in the warning. Do you still want to continue setting up?</string>
+    <string name="confirmation">Confirmation</string>
+    <string name="yes_continue">Yes, continue</string>
```
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
    <string name="yes_continue">Continue without locking</string>
```

File: res/values/strings.xml [hunk 21]
```diff
@@ -38,4 +38,29 @@
...
+    <string name="no_reboot_to_bootloader">No, reboot to bootloader</string>
+    <string name="yes_disable_now">Yes, disable now</string>
```
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
    <string name="yes_disable_now">Disable</string>
```

File: res/values/strings.xml [hunk 22]
```diff
@@ -38,4 +38,29 @@
...
+    <string name="no_disable_later">No, I will disable it later</string>
```
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
    <string name="no_disable_later">Skip for now</string>
```

File: res/values/strings.xml [hunk 23]
```diff
@@ -38,4 +38,29 @@
...
+    <string name="oem_unlock_disabling_suggestion">OEM unlocking is enabled, we recommend disabling it.
+        Would you like to disable now?
+        You can also disable later from developer options in Settings app.</string>
+    <string name="lock_your_bootloader">Lock your bootloader</string>
+    <string name="oem_unlock_fullscreen_desc">
+        The bootloader is unlocked and software integrity
+        can not be guaranteed. Any data stored on this device may be available to attackers.
+        Do not store any sensitive data on this device.\n\nLocking the bootloader at a later stage
+        will wipe all the data on this device. To keep the data safe and secure we advice you to
+        lock the bootloader before setting up the device.\n\nTo lock your bootloader, press
```
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
        lock the bootloader before setting up the device.\n\nTo lock your bootloader, visit
```

File: res/values/strings.xml [hunk 24]
```diff
@@ -38,4 +38,29 @@
...
+        \"Reboot to bootloader\" and visit this link on another device for instructions:
```
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
        the link below on another device for instructions and then press \"Reboot to bootloader\":
```

File: res/values/strings.xml (hunk 23 above)
Comment (jiteshsingh) [reply res/values/strings.xml:59]: this changes the meaning. pls recheck @matchboxbananasynergy

File: res/values/strings.xml [hunk 25]
```diff
@@ -38,4 +38,29 @@
     <string name="swipe_gestures_title">Swipe to navigate your phone</string>
     <string name="swipe_gestures_desc">Learn gestures to go home, go back and switch between apps</string>
     <string name="try_it">Try it</string>
+    <string name="oem_unlocked_warning">The bootloader is unlocked and software integrity can not be guaranteed. Any data stored on this device may be available to attackers. Do not store any sensitive data on this device.</string>
+    <string name="visit_this_link_on_another_device">Visit this link on another device:</string>
+    <string name="oem_lock_url" translatable="false">grapheneos.org/UBL</string>
+    <string name="oem_unlocked_device_setup_confirmation">Bootloader is unlocked which renders the device insecure as displayed in the warning. Do you still want to continue setting up?</string>
+    <string name="confirmation">Confirmation</string>
```
Comment (matchboxbananasynergy) [inline res/values/strings.xml]:
```suggestion
    <string name="oem_unlock_header">OEM Unlocking</string>
```
